My attempt to do todo

Maria108 opened the pull request:
*Please check off the stories that have been completed in your homework submission. Feel free to add any additional notes to the PR description.*


### User stories to implement
 - [x] A user should not be able to complete a task when creating it.
 - [x] A user can filter their view by complete or incomplete tasks.
 - [x] A user's default view is filtered to show only incomplete tasks.
 - [x] A user should be able to keep multiple lists.
 - [x] A user should have access to only their tasks. (may require devise or a creative, lightweight auth system)
 - [ ] A user viewing a task should see a history of all its changes. (task audit trail)
 - [ ] A user should be able to see a history of all completed tasks and associated actions. (app audit trail)

### Cleanup tasks
- [ ] Consolidate the repeated code between the create and edit feature specs for tasks.

### Review Status
**Note to candidate:** Do not edit this section

- [x] Specs Passing
- [ ] Two approving reviews

aaackerman commented on app/controllers/tasks_controller.rb:
```diff
@@ -1,12 +1,14 @@
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    is_completed = params[:completed].present?
```
I noticed, if I send `tasks?completed=false` as params to this route I will also see completed tasks. There's no way to do this via the UI. Did you considered any other approaches to solve the filters problem?

aaackerman commented on app/views/tasks/_form.html.haml:
```diff
@@ -9,7 +9,7 @@
     =f.label :name
     =f.text_field :name
-  .field
+  .field{:class => ("hidden" if @task.new_record?)}
```
If a pesky user wanted to edit this hidden field, is there another way to ensure new tasks aren't marked as completed?

Maria108 commented:
@aaackerman Originally I created pull request with `2.3.1` version, but then rebased against latest `master` where ruby was upgraded to `2.4.2` and looks like it's `semaphoreci` bug which can't detect new version and trying to run tests against old version, closing and opening pr should fix this problem, but I'll downgrade ruby version for now in `Gemfile`

Maria108 commented:
@aaackerman since I added multiple lists feature and changed architecture of application so specs for `tasks` should be different now, should I change them to make them pass?

sethwebster commented:
> @aaackerman since I added multiple lists feature and changed architecture of application so specs for `tasks` should be different now, should I change them to make them pass?

@Maria108 That's up to you. Generally, passing specs are a plus.

sethwebster commented:
@Maria108 -- Did you want to revise your submission or shall we continue our review as-is?

Thanks!

Maria108 commented:
Hey @sethwebster, please check my submission
